src/RA_Dlg_MemBookmark.h

```diff
@@ -8,93 +8,169 @@
 class MemBookmark
 {
 public:
-    void SetDescription(const std::wstring& string) { m_sDescription = string; }
-    void SetAddress(unsigned int nVal) { m_nAddress = nVal; }
-    void SetType(unsigned int nVal) { m_nType = nVal; }
-    void SetValue(unsigned int nVal) { m_sValue = nVal; }
-    void SetPrevious(unsigned int nVal) { m_sPrevious = nVal; }
-    void IncreaseCount() { m_nCount++; }
-    void ResetCount() { m_nCount = 0; }
-
-    void SetFrozen(bool b) { m_bFrozen = b; }
-    void SetDecimal(bool b) { m_bDecimal = b; }
-
-    inline const std::wstring& Description() const { return m_sDescription; }
-    unsigned int Address() const { return m_nAddress; }
-    unsigned int Type() const { return m_nType; }
-    unsigned int Value() const { return m_sValue; }
-    unsigned int Previous() const { return m_sPrevious; }
-    unsigned int Count() const { return m_nCount; }
-
-    bool Frozen() const { return m_bFrozen; }
-    bool Decimal() const { return m_bDecimal; }
-
+    // if you're worried about speed just delete copying, the pointer is always
+    // corrupted, I don't know how you guys got it to work before. It still needs
+    // to be movable
+    MemBookmark() noexcept = default;
+    ~MemBookmark() noexcept = default;
+    MemBookmark(const MemBookmark&) = delete;
+    MemBookmark& operator=(const MemBookmark&) = delete;
+    MemBookmark(MemBookmark&&) noexcept = default;
```

Comment:
Copying is deleted so MemBookmark would be cheap to use without making it a pointer. For something to be a NullablePointer it has to be CopyContructible and CopyAssignable, this one is neither so that means it doesn't need to be a pointer.